src/modules/contacts.js

```diff
@@ -0,0 +1,397 @@
+    <span className="numbers">{props.number}</span>
+    <span className="contact__info" onClick={props.clickOne}></span>
+  </div>
+)    
+
+const Name = (props) => (
+  <p className="contact__name" onClick={props.click}>{props.name}</p>
+)
+
+const Contact = (props) => (
+  <li className="contact">
+    <Name name={props.name} click={props.click}></Name>
+    <FurtherInfo number={props.number} clickOne={props.clickOne} clickTwo={props.clickTwo}></FurtherInfo>
+  </li>
+)
+
+const AddContact = (props) => (
+  <div className="contact__add" onClick={props.click}>
+    <p>Add contact<span className="contact__add-plus"></span></p>
+  </div>
+)
+
+const InfoWindow = (props) => (
+  <div className="contact__info-window__wrapper">
+    <div className="contact__input__wrapper">
+      <span>Name:</span>
+      <input type="text" placeholder={props.name} className="contact__add-name" />
+    </div>
+    <div className="contact__input__wrapper">
+      <span>Number:</span>
+      <input type="number" placeholder={props.number} className="contact__add-name" />
+    </div>
+    <div className="contact__input__wrapper">
+      <span className="contact__add-likes-span">Likes:</span>
+      <input type="text" placeholder={props.likes} className="contact__add-likes" />
+    </div>
+    {props.message}
+    <div className="contact__add-button" onClick={() => {props.clickOne(); props.clickTwo();}}>
+      Save
+    </div>
+    <div className="contact__add-button contact__add-button--delete" onClick={() => {props.clickThree(); props.clickTwo();}}>
+      Delete
+    </div>
+  </div>
+)
+
+// Creates an array of numbers from/to
+const createNumberArray = (lowEnd, highEnd) => {
+  let arr = [];
+  while(lowEnd <= highEnd){
+    arr.push(lowEnd++);
+  }
+  return arr
+}
+
+// Check if persons birthdate matches current date
+const checkBirthDays = (person) => person.birthDay[0] == newDate.getDate() && person.birthDay[1] == (newDate.getMonth()+1);
+
+// Check if it was atleast a month since last called
+const checkLastCalled = (person) => (person.date.day - newDate.getDate()) == 0 && (newDate.getMonth() - person.date.month) == 1;
+
+// Check if year filled out
+const checkOptions = (arr) => (arr[0] == 1 &&  arr[1] == 1 && arr[2] == 1900) ? false : true;
+
+const getLikes = (arr, name) => arr.filter(obj => obj.name === name).map(obj => obj.likes);
+
+export default class Contacts extends Component{
+  constructor() {
+    super()
+    this.date = new Date();
+
+    // Defining variables
+    this.selects = document.getElementsByClassName("contact__add-day");
+    this.years = createNumberArray(1900,this.date.getFullYear());
+    this.database = firebase.database().ref('/contacts');
+    this.months = createNumberArray(1,12);
+    this.days = createNumberArray(1,31);
+    this.birthDays = [];
+    this.unCalled = [];
+    this.inputs = [];
+    this.called = "";
+
+    // Binding functions
+    this.changeContact = this.changeContact.bind(this);
```

Comment:
this is depricated. It is common to use an arrow function definition, then you do not need the binding at all

Reply:
Uprime nevim jak byste chtel pouzivat this.promena v cele classe bez nabindovani v constructoru. Muzete uvest nejaky priklad v realnem kodu, jeste jsem se s tim nesetkal, aby se nemuseli v classe bindovat funkce.

Reply:
Určitě. Hned to bude...

```
class Foo extends Component {
  handleClickButton = (event) => {
    //handle event somehow
  }

  render (
    <button onClick={this.handleClickButton}>
      My Awesome button
    </button>
  )
}
```

Více infromací lze najít tady:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions

Reply:
Tak tohle je pro me novinka, dokonce ani vyvojari u nas, kteri jsou ciste na React, o teto moznosti nemeli poneti 😄  diky